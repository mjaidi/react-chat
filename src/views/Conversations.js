import React, { Component } from "react";
import ConversationsList from "components/ConversationsList";
import PleaseLogin from "components/PleaseLogin";
import { matchPath } from "react-router-dom";

class Conversations extends Component {
  render() {
    const { isAuthenticated } = this.props.auth;
    const hideConversationsListForMobileConversations = () => {
      const match = matchPath(this.props.history.location.pathname, {
        path: "/conversations/:id",
        exact: true,
        strict: false
      });
      if (match !== null && this.props.isMobile) {
        return true;
      }
      return false;
    };

    if (isAuthenticated()) {
      return (
        <div>
          {!hideConversationsListForMobileConversations() && (
            <ConversationsList
              client={this.props.client}
              history={this.props.history}
              activeConversation={this.props.history.location.id}
            />
          )}
        </div>
      );
    }
    return <PleaseLogin />;
  }
}

export default Conversations;
