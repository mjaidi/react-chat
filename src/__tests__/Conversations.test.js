import React from "react";
import { shallow } from "enzyme";
import Conversations from "views/Conversations";
import Auth from "../auth/auth0.js";
import history from "../history";

describe("Conversations", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Conversations auth={new Auth()} history={history} />);
  });
  it("should render correctly without crashing", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render PleaseLogin component if authentication fails", () => {
    expect(component.find("PleaseLogin").exists()).toBe(true);
    expect(component.find("ConversationsList").exists()).toBe(false);
  });

  it("should render ConversationsList component if authentication succeeds", () => {
    component.setProps({ auth: { isAuthenticated: () => true } });
    expect(component.find("PleaseLogin").exists()).toBe(false);
    expect(component.find("ConversationsList").exists()).toBe(true);
  });
});
