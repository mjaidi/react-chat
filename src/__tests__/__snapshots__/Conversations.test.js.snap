// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversations should render correctly without crashing 1`] = `<PleaseLogin />`;
